foundation-extensions/campaign-standard/campaign-api-reference.md

```diff
@@ -0,0 +1,173 @@
+# Campaign API Reference
+
+This document contains usage information for the public functions and classes in `AEPCampaign`.
+
+## Static functions
+
+- [extensionVersion](#extensionVersion)
+- [registerExtension](#registerExtension)
+- [resetLinkageFields](#resetLinkageFields)
+- [setLinkageFields](#setLinkageFields)
+
+<hr />
+
+### extensionVersion
+
+Returns the running version of the AEPCampaign extension.
+
+**Signature**
+
+{% tabs %}
+{% tab title="Swift" %}
+
+```swift
+static var extensionVersion: String
+```
+
+**Example Usage**
+
+```swift
+let campaignVersion = Campaign.extensionVersion
+```
+
+{% endtab %}
+
+{% tab title="Objective-c" %}
+
+**Signature**
+
+```objc
++ (nonnull NSString*) extensionVersion;
+```
+
+**Example Usage**
+
+```objc
+NSString *campaignVersion = [AEPMobileCampaign extensionVersion];
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+<hr />
+
+### registerExtension
+
+This API no longer exists in `AEPCampaign`. Instead, the extension should be registered by calling the `registerExtensions` API in the `MobileCore`.
+
+{% tabs %}
+{% tab title="Swift" %}
+
+**Example:**
+
+```swift
+MobileCore.registerExtensions([Campaign.self, ...], {
+  // processing after registration
+})
+```
+
+{% endtab %}
+
+{% tab title="Objective-c" %}
+
+**Example:**
+
+```objc
+[AEPMobileCore registerExtensions:@[AEPMobileCampaign.class, ...] completion:^{
+  // processing after registration
+}];
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+<hr />
+
+### resetLinkageFields
+
+Clears previously stored linkage fields in the mobile SDK and triggers a Campaign rules download request to the configured Campaign server.
+
+This method unregisters any previously registered rules with the Rules Engine and clears cached rules from the most recent rules download.
+
+{% tabs %}
+{% tab title="Swift" %}
+
+**Signature**
+
+```swift
+static func resetLinkageFields()
+```
+
+**Example Usage**
+
+```swift
+Campaign.resetLinkageFields()
+```
+
+{% endtab %}
+
+{% tab title="Objective-c" %}
+
+**Signature**
+
+```objc
++ (void) resetLinkageFields;
+```
+
+**Example Usage**
+
+```objc
+[AEPMobileCampaign resetLinkageFields];
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+<hr />
+
+### setLinkageFields
+
+Sets the Campaign linkage fields (CRM IDs) in the mobile SDK to be used for downloading personalized messages from Campaign.
+
+The set linkage fields are stored as a base64 encoded JSON string in memory and they are sent in a custom HTTP header 'X-InApp-Auth'
```

Comment:
Two questions:

1. Should the custom header be capitalized like that? (IE: It's not `x-inapp-auth`?)
2. We should wrap the header in backticks, rather than single quotes.

Reply:
The custom header is capitalized in that way. I will enclose it it backticks though.

Reply:
Got it! That makes sense to me then - just enclose it within the backticks.